Cognite.Common/PeriodicScheduler.cs

```diff
@@ -98,16 +99,17 @@ public PeriodicScheduler(CancellationToken token)
         /// <param name="operation">Function to call on each iteration</param>
         /// <param name="runImmediately">True to execute the periodic task immediately, false to first
         /// wait until triggered by interval or manually</param>
+        /// <param name="dynamic">Whether the interval is dynamic or not, e.g. cron expression</param>
         public void SchedulePeriodicTask(string? name, ITimeSpanProvider interval,
-            Func<CancellationToken, Task> operation, bool runImmediately = true)
```

Comment:
Ah, no, this isn't pleasant and won't really work. You are expecting users to have internal knowledge of issues in the scheduler.

I think it might be better to see if you can fix this by just modifying `CronTimeSpanWrapper`. See if this is fixed by just adding a small offset to `DateTime.UtcNow` when getting the next occurence, to add a tiny bit of tolerance. I.e. add one second or something. Line 111 in TimeSpanWrapper.cs